### PR TITLE
Better support for multiple RN version in test project + updated example projects

### PR DESCRIPTION
**Pull request by rotemmiz**

### CODE REVIEW

**Comment by LeoNatan** [comment]
Checks failed. Rejected! 😜

**Comment by rotemmiz** [comment]
Someone is extra joly today 🤪

File: examples/demo-react-native/ios/example.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -317,6 +387,8 @@
 			children = (
 				139FDEF41B06529B00C62182 /* libRCTWebSocket.a */,
 				392FDE331E2E5F680058768A /* libRCTWebSocket-tvOS.a */,
+				461DC7DB2047304200C366C7 /* libfishhook.a */,
```
Comment (LeoNatan) [inline examples/demo-react-native/ios/example.xcodeproj/project.pbxproj]: Was this added by RN?

File: examples/demo-react-native/ios/example.xcodeproj/project.pbxproj [hunk 2]
```diff
@@ -346,6 +418,14 @@
 				392FDE451E2E5F680058768A /* libcxxreact.a */,
 				392FDE471E2E5F680058768A /* libjschelpers.a */,
 				392FDE491E2E5F680058768A /* libjschelpers.a */,
+				461DC7EF2047304200C366C7 /* libjsinspector.a */,
```
Comment (LeoNatan) [inline examples/demo-react-native/ios/example.xcodeproj/project.pbxproj]: 🤨

Comment (LeoNatan) [reply examples/demo-react-native/ios/example.xcodeproj/project.pbxproj]: I wonder what that is.

**Comment by LeoNatan** [comment]
Approved 😂

**Pull request merged**